src/main/java/Duke.java

```diff
@@ -1,10 +1,172 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui = new Ui();
+    private static final String filePath = "./data/duke.txt";
```

Comment:
Since it is a constant, may I suggest this change?
```suggestion
    private static final String FILE_PATH = "./data/duke.txt";
```